debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerStreamingChangeEventSource.java

```diff
@@ -307,15 +308,18 @@ private void migrateTable(SqlServerPartition partition, final Queue<SqlServerCha
             throws InterruptedException, SQLException {
         final SqlServerChangeTable newTable = schemaChangeCheckpoints.poll();
         LOGGER.info("Migrating schema to {}", newTable);
-        Table tableSchema = metadataConnection.getTableSchemaFromTable(newTable);
+        Table tableSchema = metadataConnection.getTableSchemaFromTable(partition.getDatabaseName(), newTable);
         dispatcher.dispatchSchemaChangeEvent(newTable.getSourceTableId(),
                 new SqlServerSchemaChangeEventEmitter(partition, offsetContext, newTable, tableSchema,
                         SchemaChangeEventType.ALTER));
         newTable.setSourceTable(tableSchema);
     }
 
-    private SqlServerChangeTable[] processErrorFromChangeTableQuery(SQLException exception, SqlServerChangeTable[] currentChangeTables) throws Exception {
-        final Matcher m = MISSING_CDC_FUNCTION_CHANGES_ERROR.matcher(exception.getMessage());
+    private SqlServerChangeTable[] processErrorFromChangeTableQuery(String databaseName, SQLException exception,
+                                                                    SqlServerChangeTable[] currentChangeTables)
+            throws Exception {
+        final String pattern = MISSING_CDC_FUNCTION_CHANGES_ERROR.replace("[#db]", databaseName);
```

Comment:
We could still keep this a pre-compiled pattern and use it via Pattern.matcher().replaceFirst(...)?

Reply:
Not sure I understand the idea. `replaceFirst()` would return a string with a pattern replaced but we need a pattern specific for the current database name (only available at runtime).

I reworked it in a bit different way.